src/guidellm/backend/deepsparse/backend.py

```diff
@@ -1,10 +1,12 @@
-from typing import Any, AsyncGenerator, List, Optional
+import os
+from typing import Any, AsyncGenerator, Dict, List, Optional
 
-from deepsparse import Pipeline
+from deepsparse import Pipeline, TextGeneration
```

Comment:
Does this mean deepsparse is a dep of guidellm? We should keep it optional at most IMO, so could this be in a try catch with an informational message to install?

Reply:
@mgoin Could you please check [this file](https://github.com/neuralmagic/guidellm/pull/29/files#diff-86e6dcc39e1c774e18c8e3a99a1e7c7f63399a786842ca8a6b2cba48e2c270aa)? Is it kind of a validation you are talking about?

Reply:
So, since the `backend/deepsparse/__init__.py` exists we determine it as a module so before you reach the `deepsparse/backend.py` you are going to run such a validation.

Also, all the imports are like `from guidellm.backend import Backend, BackendEngine` but if you need the deepsparse just `from .guidellm.backend.deepsparse import DeepsparseBackend` which runs this validation.

Also the `deepsparse` is an optional dependency. At least [it is in optional section](https://github.com/neuralmagic/guidellm/pull/29/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)